spring-integration-redis/src/test/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducerTests.java

```diff
@@ -277,6 +366,20 @@ ReactiveRedisStreamMessageProducer reactiveRedisStreamProducer() {
 			return messageProducer;
 		}
 
+		@Bean
+		public IntegrationFlow errorHandlingFlow(ReactiveRedisTemplate<String, ?> reactiveStreamOperations) {
+			return IntegrationFlows.from("errorChannel")
+									.handle(message -> {
+										reactiveStreamOperations.opsForStream()
+																.acknowledge("testGroup",
+																((MapRecord) ((ErrorMessage) message)
+																.getOriginalMessage().getPayload()))
```

Comment:
We probably need to think about adding all of those headers for the record into a failed message we send with an error, so end-user would still be able to call simple `acknowledge()` from the `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK` header and also get all the useful information from other headers.

I also wouldn't advertise a `MapRecord` over here as an entity to deal with, but more its super, plain `Record<S, V>` interface. Also, see those raw types warnings: better to have it as `Record<?, ?>`.

OK. You ack here in the error handling, but I don't see a verification of this fact in the test-case.
Plus it is not clear how there could be a pending in the stream since you ack it here.
Isn't there going to be some race condition? Or our ack assumption is wrong?